src/main/java/net/minestom/server/raw_data/DataInitializer.java

```diff
@@ -0,0 +1,319 @@
+            Block b = Registry.BLOCK_REGISTRY.get(blockId);
+            // CASE: Returning the defaulted value, however make sure to include the defaulted data itself.
+            // to clarify: We return AIR if it doesn't exist, but we also need Air in the HashMap
+            // Therefore we check the id for this functionality.
+            if (b == Block.AIR && !blockId.equals(Block.AIR.getId())) {
+                // This should honestly never happen as the values in the Registry are based on the json file.
+                continue;
+            }
+
+            RawBlockData blockData = b.getBlockData();
+
+            blockData.explosionResistance = blockJson.get("explosionResistance").getAsDouble();
+            blockData.friction = blockJson.get("friction").getAsDouble();
+            blockData.speedFactor = blockJson.get("speedFactor").getAsDouble();
+            blockData.jumpFactor = blockJson.get("jumpFactor").getAsDouble();
+            blockData.item = Registry.MATERIAL_REGISTRY.get(blockJson.get("itemId").getAsString());
+
+            JsonArray blockStatesJson = blockJson.get("states").getAsJsonArray();
+            for (JsonElement jsonElement2 : blockStatesJson) {
+                JsonObject blockStateJson = jsonElement2.getAsJsonObject();
+                short stateId = blockStateJson.get("id").getAsShort();
+
+                // Get correct Blockstate
+                BlockState bs = Registry.BLOCK_STATE_REGISTRY.get(stateId);
+                // CASE: Returning the defaulted value, however make sure to include the defaulted data itself.
+                if (bs == BlockState.AIR_0 && stateId != BlockState.AIR_0.getId()) {
+                    // This should honestly never happen as the values in the Registry are based on the json file.
+                    continue;
+                }
+
+                RawBlockStateData blockStateData = bs.getBlockStateData();
+
+                blockStateData.destroySpeed = blockStateJson.get("destroySpeed").getAsDouble();
+                blockStateData.lightEmission = blockStateJson.get("lightEmission").getAsInt();
+                blockStateData.occluding = blockStateJson.get("doesOcclude").getAsBoolean();
+                blockStateData.pushReaction = blockStateJson.get("pushReaction").getAsString();
+                blockStateData.blocksMotion = blockStateJson.get("blocksMotion").getAsBoolean();
+                blockStateData.flammable = blockStateJson.get("isFlammable").getAsBoolean();
+                blockStateData.liquid = blockStateJson.get("isLiquid").getAsBoolean();
+                blockStateData.replaceable = blockStateJson.get("isReplaceable").getAsBoolean();
+                blockStateData.solid = blockStateJson.get("isSolid").getAsBoolean();
+                blockStateData.solidBlocking = blockStateJson.get("isSolidBlocking").getAsBoolean();
+                blockStateData.mapColor = MapColors.values()[blockStateJson.get("mapColorId").getAsInt()];
```

Comment:
I personally think we should deprecate the `.values()` method and mark it as for removal. This is creating and allocating a whole new array each time which is way too much work.